nexus/src/app/sagas/mod.rs

```diff
@@ -400,3 +401,18 @@ where
     )
     .await
 }
+
+/// Reliable persistent workflows can request that sagas be run as part of their
+/// activation by sending a SagaRequest through a supplied channel to Nexus.
+pub enum SagaRequest {
+    #[cfg(test)]
+    TestOnly,
+}
```

Comment:
The intention is for this enum to have a variant for each saga that could be requested, along with the params:

```rust
    DiskCreate { params: disk_create::Params },
```

and for Nexus to simply be unwrapping this:

```rust
match saga_request {
    ....
    DiskCreate { params: disk_create::Params } => {
        self.execute_saga::<sagas::disk_create::SagaDiskCreate>(params).await?;
    }
    ....
}
```

Unfortunately I don't have a task that uses this along with this PR: `RegionReplacementDetector` is currently a stub until physical disks have a state that this background task can query for, I simply wanted to get this work in separately to make subsequent PRs smaller :)

I believe it could help this PR make sense to wait for physical disks to have this state, and to have an implementation to go along with this.

Reply:
I see. So the TL;DR is that this PR provides a mechanism for background tasks to access `Arc<Nexus>`, more-or-less, without them storing a bunch of https://doc.rust-lang.org/std/sync/struct.Weak.html ? And we use channels with requests to Nexus to ask it to execute each saga?

That makes sense!